pkg/scaler/general.go

```diff
@@ -1014,6 +1034,12 @@ func (a *GeneralController) getUnableComputeReplicaCountCondition(gpa *autoscali
 // outdated events to be replaced were marked as outdated in the `markScaleEventsOutdated` function
 func (a *GeneralController) storeScaleEvent(behavior *autoscaling.GeneralPodAutoscalerBehavior,
 	key string, prevReplicas, newReplicas int32) {
+	//add lock
+	a.scaleUpEventsLock.Lock()
+	defer a.scaleUpEventsLock.Unlock()
+	a.scaleDownEventsLock.Lock()
+	defer a.scaleDownEventsLock.Unlock()
+
```

Comment:
I would call Lock/Unlock when necessary rather than calling this when we enter this func.

e.g add

	a.scaleUpEventsLock.Lock()
	defer a.scaleUpEventsLock.Unlock()

when `newReplicas > prevReplicas`.

while call 

```	
a.scaleDownEventsLock.Lock()
defer a.scaleDownEventsLock.Unlock()
```

when not satisfied.

Reply:
Thank you for your reply. 
In the function, if...else... are both read and write events, so  calling this func at the entry will be simpler.
![image](https://user-images.githubusercontent.com/6057029/125262196-8f7bec80-e334-11eb-89fe-dde696cea24c.png)